tokio-util/tests/future.rs

```diff
@@ -0,0 +1,255 @@
+use std::{
+    future::{pending, ready, Future},
+    task::{Context, Poll},
+};
+
+use futures_test::task::new_count_waker;
+use tokio::pin;
+use tokio_test::{assert_pending, assert_ready_eq};
+use tokio_util::{future::FutureExt, sync::CancellationToken};
+
+#[test]
+fn ready_future_with_cancellation_token_test() {
+    let (waker, _) = new_count_waker();
+    let token = CancellationToken::new();
+
+    let ready_fut = ready(());
+
+    let ready_with_token_fut = ready_fut.with_cancellation_token(&token);
+
+    pin!(ready_with_token_fut);
+
+    let res = ready_with_token_fut
+        .as_mut()
+        .poll(&mut Context::from_waker(&waker));
+
+    assert_ready_eq!(res, Some(()));
+}
+
+#[test]
+fn pending_future_with_cancellation_token_test() {
+    let (waker, _) = new_count_waker();
+    let token = CancellationToken::new();
+
+    let pending_fut = pending::<()>();
+
+    let pending_with_token_fut = pending_fut.with_cancellation_token(&token);
+
+    pin!(pending_with_token_fut);
+
+    let res = pending_with_token_fut
+        .as_mut()
+        .poll(&mut Context::from_waker(&waker));
+
+    assert_pending!(res);
+}
+
+#[test]
+fn pending_fut_with_already_cancelled_token_test() {
```

Comment:
We also want a `ready_fut_with_already_cancelled_token_test` to cover the https://github.com/tokio-rs/tokio/pull/7475#discussion_r2229891851.